### PR TITLE
fixed error when search result has results in different currencies

### PR DESCRIPTION
**Pull request by DFSko**
resolves happy-travel/agent-app-project#822

### CODE REVIEW

File: Api/Services/Accommodations/Availability/Steps/RoomSelection/RoomSelectionPriceProcessor.cs [hunk 1]
```diff
@@ -18,37 +18,66 @@ public RoomSelectionPriceProcessor(IPriceProcessor priceProcessor)
...
+        public async Task<Result<SingleAccommodationAvailability, ProblemDetails>> ConvertCurrencies(SingleAccommodationAvailability availabilityDetails)
+        {
+            var convertedRoomContractSets = new List<RoomContractSet>(availabilityDetails.RoomContractSets.Count);
+            foreach (var roomContractSet in availabilityDetails.RoomContractSets)
+            {
+                var (_, isFailure, convertedRoomContractSet, error) = await _priceProcessor.ConvertCurrencies(roomContractSet,
+                    ProcessPrices,
+                    GetCurrency);
+                    
+                if (isFailure)
```
Comment (konstyefremov) [inline Api/Services/Accommodations/Availability/Steps/RoomSelection/RoomSelectionPriceProcessor.cs:30]: Maybe it's worth checking all contracts first and returning the combined error?

**Review by konstyefremov — APPROVED** [review]

File: Api/Services/Accommodations/Availability/Steps/RoomSelection/RoomSelectionPriceProcessor.cs [hunk 2]
```diff
@@ -18,37 +18,66 @@ public RoomSelectionPriceProcessor(IPriceProcessor priceProcessor)
...
             return new SingleAccommodationAvailability(availabilityId: source.AvailabilityId,
                 checkInDate: source.CheckInDate,
                 roomContractSets: roomContractSets,
                 htId: source.HtId,
                 countryHtId: source.CountryHtId,
                 localityHtId: source.LocalityHtId);
         }
+        
+        
+        private static async ValueTask<RoomContractSet> ProcessPrices(RoomContractSet roomContractSet, PriceProcessFunction function)
```
[the remark this answers is not in the document]
Comment (DFSko) [reply Api/Services/Accommodations/Availability/Steps/RoomSelection/RoomSelectionPriceProcessor.cs]: Fixed

File: Api/Services/Accommodations/Availability/Steps/WideAvailabilitySearch/WideAvailabilityPriceProcessor.cs [hunk 3]
```diff
@@ -83,35 +117,16 @@ public List<AccommodationAvailabilityResult> AlignPrices(List<AccommodationAvail
...
-                checkInDate: accommodationAvailability.CheckInDate,
-                checkOutDate: accommodationAvailability.CheckOutDate,
-                htId: accommodationAvailability.HtId,
-                supplierAccommodationCode: accommodationAvailability.SupplierAccommodationCode,
-                countryHtId: accommodationAvailability.CountryHtId,
-                localityHtId: accommodationAvailability.LocalityHtId);
-            
-            return convertedAccommodationAvailability;
-        }
+        private static async ValueTask<RoomContractSet> ProcessPrices(RoomContractSet roomContractSet, PriceProcessFunction function)
```
[the remark this answers is not in the document]
Comment (DFSko) [reply Api/Services/Accommodations/Availability/Steps/WideAvailabilitySearch/WideAvailabilityPriceProcessor.cs]: Fixed

File: Api/Services/Accommodations/Availability/Steps/RoomSelection/RoomSelectionPriceProcessor.cs (hunk 1 above)
Comment (DFSko) [reply Api/Services/Accommodations/Availability/Steps/RoomSelection/RoomSelectionPriceProcessor.cs:30]: I guess all errors will be the same and return first faster

**Pull request merged**